pkg/health/condition/check_backup_ready.go

```diff
@@ -37,101 +38,194 @@ const (
+		return handleOnlyDeltaSnapshotLeaseRenewedCase(
+			fullSnapshotLeaseCreationTime.Time, fullSnapshotDuration,
+			deltaSnapshotLeaseRenewTime.Time, 2*etcd.Spec.Backup.DeltaSnapshotPeriod.Duration,
+		)
+	}
+
+	// Most probable during a startup of a new cluster and only full snapshot has been taken
+	if fullSnapshotLeaseRenewTime != nil && deltaSnapshotLeaseRenewTime == nil {
+		return handleSnapshotterStartupCase(
+			fullSnapshotLease.Spec.RenewTime.Time,
+			5*etcd.Spec.Backup.DeltaSnapshotPeriod.Duration,
+		)
+	}
+
+	// Both snap leases are renewed, ie, maintained. Both are expected to be renewed periodically
+	return handleRenewedSnapshotLeases(
+		fullSnapshotLease.Spec.RenewTime.Time, fullSnapshotDuration,
+		deltaSnapSnapshotLease.Spec.RenewTime.Time, 2*etcd.Spec.Backup.DeltaSnapshotPeriod.Duration,
+	)
+}
+
+func (a *backupReadyCheck) fetchLease(ctx context.Context, name string, namespace string) (*coordinationv1.Lease, error) {
+	lease := &coordinationv1.Lease{}
+	err := a.cl.Get(ctx, types.NamespacedName{Name: name, Namespace: namespace}, lease)
+	return lease, err
+}
+
+func createBackupConditionResult(status druidv1alpha1.ConditionStatus, reason string, message string) *result {
+	return &result{
+		conType: druidv1alpha1.ConditionTypeBackupReady,
+		status:  status,
+		reason:  reason,
+		message: message,
+	}
+}
+
+func isLeaseStale(renewTime time.Time, renewalGracePeriod time.Duration) bool {
+	return time.Since(renewTime) > renewalGracePeriod
+}
+
+func wasLeaseCreatedRecently(creationTime time.Time, creationGracePeriod time.Duration) bool {
+	return time.Since(creationTime) < creationGracePeriod
+}
+
+// handleOnlyDeltaSnapshotLeaseRenewedCase handles cases where only delta snapshot lease is renewed,
+// while the full snapshot lease is not renewed, possibly since it might have been recreated recently.
+// Treats backup as succeeded if delta snapshot lease is renewed within the required time window
+// and full snapshot lease object is not older than the computed full snapshot duration.
+func handleOnlyDeltaSnapshotLeaseRenewedCase(fullSnapshotLeaseCreationTime time.Time, fullSnapshotLeaseCreationGracePeriod time.Duration, deltaSnapshotLeaseRenewTime time.Time, deltaSnapshotLeaseRenewalGracePeriod time.Duration) *result {
+	// wasFullSnapshotLeaseCreatedRecently indicates whether full snapshot was created within the given grace period. If it was,
+	// then it can be renewed only upon the next scheduled full snapshot event, so until then we cannot assume the status of the last
+	// full snapshot. But if it was created before the grace period and not been renewed, then full snapshot lease can be considered stale.
+	wasFullSnapshotLeaseCreatedRecently := wasLeaseCreatedRecently(fullSnapshotLeaseCreationTime, fullSnapshotLeaseCreationGracePeriod)
+
+	// isDeltaSnapshotLeaseStale indicates whether the delta snapshot lease was renewed within the given grace period.
```

Comment:
comment says `indicates whether the delta snapshot lease was renewed within the given grace period.` whereas what you actually check is if the delta snapshot lease is stale (opposite of what the comment says). Can you correct the comment?